rapidast.py

```diff
@@ -444,5 +472,49 @@ def run():
         sys.exit(0)
 
 
+def merge_sarif_files(directory: str, properties: dict, output_filename: str):
+    """
+    Searches for .sarif and .sarif.json files in a directory, merges them,
+    and adds "properties" as properties to the merged SARIF log
+
+    Args:
+        directory: The directory to search for SARIF files
+        properties: A dictionary containing arbitrary properties for each scanner
+        output_filename: The name of the output merged SARIF file
+    """
+    merged_runs = []
+    for filename in os.listdir(directory):
```

Comment:
Merged_runs are not extended properly. The 'directory' points to the root result file and the SARIF files are under a directory like 'zap', 'garak', etc.

Reply:
Ah, that's right, we're not listing the files recursively. Let me fix that

Otoh, I'm not sure what you mean by "Merged_runs are not extended properly". Could you clarify?

Reply:
I just meant that  the line`merged_runs.extend(data["runs"])` would not be called because no sarif file is found.

Reply:
Got it ! I’ve updated the code so it now iterates through all the subdirectories